### ISSUE TITLE
How to further distill the RadioV4 model？

### ISSUE THREAD

**Issue by Pixel16Bit** [NVlabs/RADIO#178]
I now want to further distill the RadioV4 model onto myself dataset. I am unsure how to obtain the intermediate-layer QK information or other features from RadioV4. Is RadioV4 based on the standard ViT architecture? If I load the model using the standard ViT structure, can I access the attention information from the intermediate layers?   
 def forward(self, x: torch.Tensor):
        B = x.shape[0]
        x = self.patch_embed(x)
        cls_token = self.cls_token + self.pos_embed[:, :1, :]
        x = x + self.pos_embed[:, 1:, :]
        x = torch.cat((cls_token.expand(B, -1, -1), x), dim=1)
        for blk in self.blocks:
            x = blk(x)
        x = self.norm(x)
        return x[:, 0]
like this？

**Comment by mranzinger** [comment]
Hello, here's the forward implementation for the model:
https://github.com/NVlabs/RADIO/blob/main/radio/enable_cpe_support.py#L26-L33

Note that there is one layer of wrapping being done, with the model being returned by the `torch.hub.load(...)` being a `RADIOModel` instance. It has a `.model` member, which is a TIMM [VisionTransformer](https://github.com/huggingface/pytorch-image-models/blob/main/timm/models/vision_transformer.py#L681) instance.

Glancing at the code, I think you may ultimately need to monkey patch the [Attention](https://github.com/huggingface/pytorch-image-models/blob/main/timm/layers/attention.py#L18) forward pass, since it doesn't look like it exposes a way to get the attention weights. I'm optimistic that your AI agent of choice could make quick work of this.
